### ISSUE TITLE
No audio output

### ISSUE THREAD

**Issue by Siegz** [DuckBoss/JJMumbleBot#186]
I am not getting any audio output from the youtube/soundboard plugin. Even though i have installed all the dependencies, and the log doesn't seem to show any errors. Could this be something vlc related?

![kuva](https://user-images.githubusercontent.com/10603082/76624979-fc178900-653e-11ea-8aaa-55d178eea954.png)

**Comment by DuckBoss** [comment]
Sorry for the late response I've been extremely busy.
I posted a comment regarding this issue to another user, maybe it will help you.
___________________________________________________________________________________________
The VLC problem where it doesn't play audio from youtube is usually a problem with youtube constantly changing their infrastructure and protecting from unauthorized video/audio extraction.
This is resolved by making sure your youtube-dl python package is up to date and your 'youtube.lua' file is the most recently updated one from the VLC github link:
https://github.com/videolan/vlc/blob/master/share/lua/playlist/youtube.lua
(this file is placed in the vlc/share/lua/playlist/ directory)

If the problem still occurs, it usually means that the youtube-dl python package hasn't been updated to resolve the changes on youtube's side.

**Comment by MacLemon** [comment]
This is another data point, just to save others some time in case.

Same problem here. Thanks for posting the possible workarounds to try @DuckBoss!

`youtube-dl` is at version 2020.03.24 (which is current as of this posting according to [https://yt-dl.org/](https://yt-dl.org/))
I've also tried replacing `youtube.lua` in my `VLC media player 3.0.8 Vetinari (revision 3.0.8-0-gf350b6b5a7)` [from the repository](https://github.com/videolan/vlc/blob/master/share/lua/playlist/youtube.lua).

Both measures *did not change* the behaviour to solve the problem of Youtube videos not playing any sound in Mumble. I'm afraid we have to at least wait until the next `youtube-dl` update before we get a chance of having this work again.

**Comment by MacLemon** [comment]
I also don't get any sound when using a `.wav` file for the soundboard.

This is the audio file that I'm trying to use.
```
$ file rickroll.wav
rickroll.wav: RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, stereo 48000 Hz
```

JJMumbleBot will successfully list the file, but not play it.

`!sb 'rickroll'` according to the Wiki does not find the file `!sb rickroll` (without single quotes) does.

The log tells me
```
Message Received: [MacLemon -> !sb rickroll]
INFO:RuntimeLogging:Commands Received: [MacLemon -> !sb rickroll]
VLC media player 3.0.8 Vetinari (revision 3.0.8-0-gf350b6b5a7)
```
but there is no sound playing and my Mumble client also doesn't show any audio activity on the JJMumbleBot Icon in the list.

Am I doing this right? Should a `.wav` file with the specs mentioned above (converted via `ffmpeg`) play with JJMumbleBot? (It does play with VLC on macOS.) The server I'm running the bot on doesn't have a sound card obviously. :-)
If that file *should* actually play, then maybe there's a problem with `vlc` integration, rather than `youtube-dl`.

## Version info
- JJMumblebot is v2.3.1 (cloned via `git`)
- Python 3.7.7
- VLC media player 3.0.8 Vetinari (revision 3.0.8-0-gf350b6b5a7)

Anything else I can provide to help debugging this?

**Comment by DuckBoss** [comment]
Hey @MacLemon I recently resolved the youtube issue.
The solution was not a problem with JJMumbleBot, but a problem with VLC and youtube's video copyrights blocking certain videos from playing. Since VLC complies with this, it is unable to play some youtube videos (although I've noticed it works sometimes and not other times).
**Unfortunately, this is out of my hands and I will try to implement something in v3.0.0 to display an error and continue running instead of crashing the bot.**

The following worked for me to make the youtube plugin work for many videos:
1) Update the youtube.lua file with the latest one from here:
https://github.com/videolan/vlc/blob/master/share/lua/playlist/youtube.lua
2) Keep youtube-dl updated to the newest build.
